argparse.c

```diff
@@ -47,10 +47,11 @@ argparse_error(struct argparse *self, const struct argparse_option *opt,
 }
 
 static int
-argparse_getvalue(struct argparse *self, const struct argparse_option *opt,
+argparse_getvalue(struct argparse *self, struct argparse_option *opt,
                   int flags)
 {
     const char *s = NULL;
+    opt->parsed = 1;
```

Comment:
why setting this for all arguments?

Reply:
Because I thought it would be the simplest solution to the problem. You set it to 1 for arguments that are used at the command line and at the end of parsing you simply check if the required options have been parsed (meaning used in this context).